### PR TITLE
docs: describe array/slice behaviour

### PR DESCRIPTION
**Pull request by UweKrueger**
This PR documents how slices work (since #10219) and adds further test cases to avoid regressions.

I've somewhat re-arranged and extended the documentation for arrays in general to make it a little bit more consecutive.

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -610,20 +610,39 @@ If you do not specify the type explicitly, by default float literals
 will have the type of `f64`.
 
 ### Arrays
-
+#### Basic Array Concepts
```
Comment (medvednikov) [inline doc/docs.md:613]: I would just keep "Arrays" for this section.

File: doc/docs.md [hunk 2]
```diff
@@ -768,9 +815,9 @@ users.sort(a.name > b.name) // reverse sort by User.name string field
 
 #### Array Slices
 
-Slices are partial arrays. They represent every element between two indices
-separated by a .. operator. The right-side index must be greater than or equal
-to the left side index.
+A slices is a part of a parent array. Initially it refers to the elements
```
Comment (spytheman) [inline doc/docs.md:818]: `A slice`

File: doc/docs.md (hunk 1 above)
Comment (medvednikov) [reply doc/docs.md:613]: Ah sorry I misread it as a replacement, but it's a new subsection.

It's good.

File: doc/docs.md [hunk 3]
```diff
@@ -782,14 +829,47 @@ println(nums[..4]) // [0, 10, 20, 30]
...
 
 ```v
 array_1 := [3, 5, 4, 7, 6]
 mut array_2 := [0, 1]
 array_2 << array_1[..3]
-println(array_2) // [0, 1, 3, 5, 4]
+println(array_2) // `[0, 1, 3, 5, 4]`
+```
+
+A slices is always created with the smallest possible capacity `cap == len` (see
```
Comment (spytheman) [inline doc/docs.md:843]: `A slice`

Comment (medvednikov) [inline doc/docs.md:843]: A slice

File: doc/docs.md [hunk 4]
```diff
@@ -815,10 +815,9 @@ users.sort(a.name > b.name) // reverse sort by User.name string field
 
 #### Array Slices
 
-Slices are arrays that are created an alternative view to a part of a parent array.
-They represent the elements between two indices
-separated by a `..` operator. The right-side index must be greater than or equal
-to the left side index.
+A slices is a part of a parent array. Initially it refers to the elements
```
Comment (JalonSolov) [inline doc/docs.md]: "A slice" rather than "A slices"

Comment (UweKrueger) [reply doc/docs.md]: already corrected...

**Pull request merged**